### PR TITLE
Fix for https://github.com/dokufreaks/plugin-tag/issues/252

### PR DESCRIPTION
**Pull request by Miro-Collas**
Fix for #252

### CODE REVIEW

**Comment by nerun** [comment]
Thx for this fix!

File: helper.php [hunk 1]
```diff
@@ -552,10 +552,20 @@ protected function replacePlaceholders($tag) {
         $user = $INPUT->server->str('REMOTE_USER');
 
         //only available for logged-in users
-        if($USERINFO) {
-            $name  = cleanID($USERINFO['name']);
+        if(isset($USERINFO)) {
+            if(is_array($USERINFO) && isset($USERINFO['name'])) {
```
Comment (Klap-in) [inline helper.php:556]: As far as I know, you can directly check `isset($USERINFO['name'])` or even an sub item without the other checks.

Comment (Miro-Collas) [reply helper.php:556]: > As far as I know, you can directly check `isset($USERINFO['name'])` or even an sub item without the other checks.

Call it being overly careful.

Comment (fiwswe) [reply helper.php:556]: @Miro-Collas You may want to read the [documentation](https://www.php.net/manual/en/function.isset) ;-) What @Klap-in wrote is officially documented to work. So it's unlikely to change in some (minor) PHP update.

Comment (Miro-Collas) [reply helper.php:556]: > @Miro-Collas You may want to read the [documentation](https://www.php.net/manual/en/function.isset) ;-) What @Klap-in wrote is officially documented to work. So it's unlikely to change in some (minor) PHP update.

That snark was really uncalled for. I never said @Klap-in was wrong, I said I was being "overly careful" - or "yes, that is probably overkill". However, the changes don't break the plugin, don't change the logic, and don't hurt anything.

If you're not happy with the changed code, by all means fix #252 as you see fit, test, submit your own PR.

Comment (fiwswe) [reply helper.php:556]: @Miro-Collas I apologise. Misunderstandings abound. My comment was not meant as a snark, hence the smiley. And I never claimed that you said that @Klap-in was wrong.

**Comment by Klap-in** [comment]
Thanks for fixing!

File: helper.php (hunk 1 above)
Comment (Miro-Collas) [reply helper.php:556]: @fiwswe Thank you. :-)
Btw, there was an actual bug in my PR, a few lines down. I fixed it a sort while ago. LOL

**Comment by nerun** [comment]
I got another error with this fix:
```
2023-09-10 15:17:26	/home2/gurpzine/public_html/wiki/lib/plugins/tag/helper.php(564)	E_WARNING: Uninitialized string offset 0
  #0 /home2/gurpzine/public_html/wiki/lib/plugins/tag/helper.php(564): dokuwiki\ErrorHandler::errorHandler()
  #1 /home2/gurpzine/public_html/wiki/lib/plugins/tag/helper.php(535): helper_plugin_tag->replacePlaceholders()
  #2 [internal function]: helper_plugin_tag->cleanTag()
  #3 /home2/gurpzine/public_html/wiki/lib/plugins/tag/helper.php(524): array_map()
  #4 /home2/gurpzine/public_html/wiki/lib/plugins/tag/helper.php(511): helper_plugin_tag->cleanTagList()
  #5 /home2/gurpzine/public_html/wiki/lib/plugins/tag/helper.php(248): helper_plugin_tag->parseTagList()
  #6 /home2/gurpzine/public_html/wiki/lib/plugins/tag/syntax/topic.php(100): helper_plugin_tag->getTopic()
  #7 /home2/gurpzine/public_html/wiki/inc/parser/renderer.php(119): syntax_plugin_tag_topic->render()
  #8 /home2/gurpzine/public_html/wiki/inc/parserutils.php(700): Doku_Renderer->plugin()
  #9 /home2/gurpzine/public_html/wiki/inc/parserutils.php(158): p_render()
  #10 /home2/gurpzine/public_html/wiki/inc/parserutils.php(96): p_cached_output()
  #11 /home2/gurpzine/public_html/wiki/inc/Ui/PageView.php(68): p_wiki_xhtml()
  #12 /home2/gurpzine/public_html/wiki/inc/Action/Show.php(37): dokuwiki\Ui\PageView->show()
  #13 /home2/gurpzine/public_html/wiki/inc/template.php(100): dokuwiki\Action\Show->tplContent()
  #14 [internal function]: tpl_content_core()
  #15 /home2/gurpzine/public_html/wiki/inc/Extension/Event.php(133): call_user_func_array()
  #16 /home2/gurpzine/public_html/wiki/inc/Extension/Event.php(199): dokuwiki\Extension\Event->trigger()
  #17 /home2/gurpzine/public_html/wiki/inc/template.php(85): dokuwiki\Extension\Event::createAndTrigger()
  #18 /home2/gurpzine/public_html/wiki/lib/tpl/dokuwiki/main.php(59): tpl_content()
  #19 /home2/gurpzine/public_html/wiki/inc/actions.php(27): include('/home2/gurpzine...')
  #20 /home2/gurpzine/public_html/wiki/doku.php(126): act_dispatch()
  #21 {main}
```

Looks better change line 563 from this:
```
if(is_array($USERINFO) && isset($USERINFO['grps'])) {
    $group = cleanID($USERINFO['grps'][0]);
}
```
To this:
```
if(is_array($USERINFO) && isset($USERINFO['grps'][0])) {
    $group = cleanID($USERINFO['grps'][0]);
}
```

**Comment by Miro-Collas** [comment]
@nerun  well spotted, and my bad for not checking that.

**Comment by Miro-Collas** [comment]
New PR created: #254

**Pull request merged**